src/erlazure_insights.erl

```diff
@@ -0,0 +1,132 @@
+-module(erlazure_insights).
+
+-export([
+        get_events/4,
+        get_time_offset_as_timestamp/2,
+        get_events_all/4,
+        follow_token/3
+    ]).
+
+-include("erlazure.hrl").
+
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-export([test_url_construction/0]).
+-endif.
+
+-type datetime() :: {{number(), number(), number},
+    {number(), number(), number()}}.
+
+-type offset() :: {seconds | minutes | hours, number()}.
+
+-spec get_events_all(string(), datetime(), offset(), #azure_config{}) ->
+    {ok, list()} | {{error, term()}, {accumulated, list()}}.
+get_events_all(SubscriptionId, To, Offset = {_Unit, _OffsetSeconds}, Config=#azure_config{}) ->
```

Comment:
Since this is a library, which is supposed to be used for general purposes, what if someone wants more sophisticated query? 

I think the `get_events` and `_all` api should not be tied to timestamp only filtering, it should accept Filter string or something like that.


And in order to make it easier for usecase we are interested in, we can provide some shortcut function to generate this Filter string.